File: bridge/Bridge.py
class Shape:
    def draw(self):
        raise NotImplementedError


class RedCircle(Shape):
    x: int
    y: int
    color: str

    def __init__(self, x, y):
        self.x = x
        self.y = y
        self.color = "red"

    def draw(self):
        print(f"Drawing a {self.color} circle at ({self.x}, {self.y})")


class BlueSquare(Shape):
    x: int
    y: int
    color: str

    def __init__(self, x, y):
        self.x = x
        self.y = y
        self.color = "blue"

    def draw(self):
        print(f"Drawing a {self.color} square at ({self.x}, {self.y})")


class Drawer:
    shape: Shape

    def __init__(self, shape: Shape):
        self.shape = shape

    def draw(self):
        self.shape.draw()


def test_bridge():
    drawer = Drawer(RedCircle(10, 10))
    drawer.draw()
    drawer = Drawer(BlueSquare(20, 20))
    drawer.draw()


if __name__ == "__main__":
    test_bridge()

File: bridge/Bridge.cpp
#include <iostream>
#include <string>

using namespace std;

class Shape
{
public:
    virtual void draw() = 0;
};

class RedCircle : public Shape
{
private:
    int x, y;
    string color;

public:
    RedCircle(int x, int y)
    {
        this->x = x;
        this->y = y;
        this->color = "red";
    }

    void draw()
    {
        cout << "Drawing a " << color << " circle at (" << x << ", " << y << ")" << endl;
    }
};

class BlueSquare : public Shape
{
private:
    int x, y;
    string color;

public:
    BlueSquare(int x, int y)
    {
        this->x = x;
        this->y = y;
        this->color = "blue";
    }

    void draw()
    {
        cout << "Drawing a " << color << " circle at (" << x << ", " << y << ")" << endl;
    }
};

class Drawer
{
private:
    Shape *shape;

public:
    Drawer(Shape *shape)
    {
        this->shape = shape;
    }

    void draw()
    {
        shape->draw();
    }
};

int main()
{
    Drawer *drawer = new Drawer(new RedCircle(10, 10));
    drawer->draw();
    drawer = new Drawer(new BlueSquare(20, 20));
    drawer->draw();
    return 0;
}
